index.js

```diff
@@ -573,7 +680,7 @@ Peer.prototype._maybeReady = function () {
 
       items.forEach(function (item) {
         // Spec-compliant
-        if (item.type === 'transport') {
+        if (item.type === 'transport' && item.selectedCandidatePairId) {
```

Comment:
I was getting rare candidates that did not have that property. I'll remove this and open a separate bug to look into it.

Reply:
I've noticed this error only happens in the new tests, so we might want to figure it out before merging:

Here's a sample candidate that causes the failure (they are somewhat rare):
```javascript
{
	bytesReceived:0,
	bytesSent:0,
	dtlsState:"new",
	id:"RTCTransport_audio_1",
	localCertificateId:"RTCCertificate_F7:92:F4:5E:99:2C:F6:66:92:6C:DA:21:D7:30:AB:D7:61:E5:E2:A2:82:AF:58:2D:1D:1D:BD:99:D4:7E:D8:92",
	timestamp:1520117393473.471,
	type:"transport"
}
``

That's the whole object. Note the absence of `localCandidateId` event though it is a transport candidate.

There IS a localCertificateId, which is what I see in the spec for the TransportStats dict. It doesn't seem localCandidateId is in the spec for TransportStats?

Reply:
<img width="454" alt="screen shot 2018-03-03 at 5 56 16 pm" src="https://user-images.githubusercontent.com/14932492/36940187-3433f060-1f0c-11e8-86f0-3cabf9714246.png">

Reply:
Nevermind, the issue is that `candidatePairs[item.selectedCandidatePairId]` is undefined.

Reply:
Interesting, basically another aspect of #215. Then workaround makes sense.

I'm just wondering why Chrome even adds `type=transport` without `selectedCandidatePairId` property, since it clearly has this property on later stage and not much useful without it.
Might be worth to report as Chrome bug and add a comment with link to bug report (haven't found existing bug report for this).

Reply:
It only happens when `dtlsState` is "new", which might suggest the selectedCandidatePairId isn't known at that point.

Reply:
I've just figured that out, dictionary only contains about half of the properties from the spec. Looks like it is necessary to check `dtlsState` property.

Reply:
Here: https://w3c.github.io/webrtc-pc/#rtcdtlstransportstate-enum